examples/memcpy/src/lib.rs

```diff
@@ -0,0 +1,27 @@
+#![no_std]
+use stellar_contract_sdk::{contractfn, Env, Binary, FixedLengthBinary};
+
+#[contractfn]
+pub fn bin_new(e: Env, len: u32) -> Binary {
+    let buf: [u8; 4] = [0,1,2,3];
+    e.binary_new_from_linear_memory(buf.as_ptr() as u32, len)
+}
+
+#[contractfn]
+pub fn from_guest(e: Env, b: Binary, b_pos: u32, lm_pos: u32, len: u32) -> Binary {
+    let buf: [u8; 4] = [0,1,2,3];
+    assert!(lm_pos + len > buf.len() as u32);
+    let lm_pos: u32 = unsafe { buf.as_ptr().add(lm_pos as usize) as u32};
+    e.binary_copy_from_linear_memory(b, b_pos, lm_pos, len)
+}
+
+#[contractfn]
+pub fn to_guest(e: Env, b: Binary, b_pos: u32, lm_pos: u32, len: u32){
+    let buf: [u8; 4] = [0; 4];    
+    assert!(lm_pos + len > buf.len() as u32); 
+    let lm_pos: u32 = unsafe { buf.as_ptr().add(lm_pos as usize) as u32};
+    e.binary_copy_to_linear_memory(b.clone(), b_pos, lm_pos, len);
+    for idx in lm_pos..buf.len() as u32 {
+        assert_eq!(buf[idx as usize], b.get(b_pos + idx));
+    }
+}
```

Comment:
Could you add some tests executing these functions and demonstrating that they work?

Reply:
Oh right... there's no way to test this outside of the wasm 🤔. Has this been tested with the CLI?

Reply:
This was tested in the host unit test: https://github.com/stellar/rs-stellar-contract-env/blob/main/stellar-contract-env-host/src/test.rs#L1485
Is it good enough? Can try CLI too if needed.